### ISSUE TITLE
Need some clarification on the way direcotries with index.js are handled

### ISSUE THREAD

**Issue by rackaam** [fastify/fastify-autoload#90]
From the doc:

> Each script file within a directory is treated as a plugin unless the directory contains an index file (e.g. index.js). In that case only the index file will be loaded.

In practice this is true only when the directory containing the index.js doesn't include other sub-directories. [In this case](https://github.com/fastify/fastify-autoload/blob/c1d31a7d72b37814f6c1214ee968625ed9d835e9/index.js#L66) the others plugins in the parent-directory and in the sub-directories are also loaded.

I don't know if its a bug or if the documentation isn't up-to-date. IMO adding a sub-directory shouldn't change the way the others plugins in the parent directory are handled.

Thanks!

**Comment by mcollina** [comment]
It seems that was just added in https://github.com/fastify/fastify-autoload/pull/87.

Why is the current docs wrong? I _guess_ the doc is not up-to-date, but it can also be somewhat a bug.

Can you please articulate more clearly with an example? It's hard to follow.

**Comment by mcollina** [comment]
Can you please check if v3.0.2 is ok? If so maybe #87 was not really correct.

**Comment by rackaam** [comment]
It seems to work fine in 3.0.2

**Comment by mcollina** [comment]
@rackaam can this be closed then?

**Comment by rackaam** [comment]
Yep, no reason to keep this open after https://github.com/fastify/fastify-autoload/pull/92
